File: python/enolib/elements/missing/missing_element_base.py
class MissingElementBase:
    def __init__(self, key, parent):
        self._key = key
        self._parent = parent

    def _missing_error(self, _element):
        self._parent._missing_error(self)

    def key(self, _loader):
        self._parent._missing_error(self)

    def optional_comment(self, _loader):
        return None

    def optional_string_comment(self):
        return None

    def raw(self):
        return None

    def required_comment(self, _loader):
        self._parent._missing_error(self)

    def required_string_comment(self):
        self._parent._missing_error(self)

    def string_key(self):
        self._parent._missing_error(self)

File: ruby/lib/enolib/elements/missing/missing_element_base.rb
# frozen_string_literal: true

module Enolib
  class MissingElementBase
    attr_reader :parent

    def initialize(key, parent)
      @key = key
      @parent = parent
    end

    def _missing_error(_element)
      @parent._missing_error(self)
    end

    def key(_loader)
      @parent._missing_error(self)
    end

    def optional_comment(_loader)
      nil
    end

    def optional_string_comment
      nil
    end

    def required_comment(_loader)
      @parent._missing_error(self)
    end

    def required_string_comment
      @parent._missing_error(self)
    end

    def string_key
      @parent._missing_error(self)
    end
  end
end
